packages/eas-cli/src/commands/env/push.ts

```diff
@@ -0,0 +1,176 @@
+import { Flags } from '@oclif/core';
+import fs from 'fs-extra';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import { EASEnvironmentFlag } from '../../commandUtils/flags';
+import {
+  EnvironmentVariableFragment,
+  EnvironmentVariableScope,
+  EnvironmentVariableVisibility,
+} from '../../graphql/generated';
+import {
+  EnvironmentVariableMutation,
+  EnvironmentVariablePushInput,
+} from '../../graphql/mutations/EnvironmentVariableMutation';
+import { EnvironmentVariablesQuery } from '../../graphql/queries/EnvironmentVariablesQuery';
+import Log from '../../log';
+import { promptAsync } from '../../prompts';
+
+export default class EnvironmentValuePush extends EasCommand {
+  static override description = 'push env file';
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  static override flags = {
+    ...EASEnvironmentFlag,
+    path: Flags.string({
+      description: 'Path to the env file',
+      default: '.env.local',
+    }),
+  };
+
+  async runAsync(): Promise<void> {
+    const {
+      flags: { environment, path: envPath },
+    } = await this.parse(EnvironmentValuePush);
+    const {
+      privateProjectConfig: { projectId },
+      loggedIn: { graphqlClient },
+    } = await this.getContextAsync(EnvironmentValuePush, {
+      nonInteractive: true,
+    });
+
+    if (!environment) {
+      throw new Error('Please provide an environment to push the env file to.');
+    }
+
+    const updateVariables: Record<string, EnvironmentVariablePushInput> =
+      await this.parseEnvFileAsync(envPath, environment);
+
+    const variableNames = Object.keys(updateVariables);
+
+    const { appVariables: existingVariables } = await EnvironmentVariablesQuery.byAppIdAsync(
+      graphqlClient,
+      { appId: projectId, environment, filterNames: variableNames }
+    );
+
+    // Check if any of the variables already exist in the environment. Prompt the user to overwrite them.
+    const existingSensitiveVariables = existingVariables.filter(
+      variable => variable.value === null
+    );
+
+    if (existingSensitiveVariables.length > 0) {
```

Comment:
```suggestion
    if (existingDifferentSensitiveVariables.length > 0) {
```
Shouldn't we only print it for different variables?